docs/spec/spec.md

```diff
@@ -0,0 +1,297 @@
+
+### Object Schema
+
+#### Spec
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| provisioner*| ProvisionerReference | The provisioner used to create any backing resources and configuration. | Immutable. |
+| arguments | runtime.RawExtension (JSON object)| Arguments passed to the provisioner for this specific source. | Arguments must validate against provisioner's parameters. |
+| channel | ObjectRef | Specify an existing channel to use to emit events. If empty, create a new Channel using the cluster/namespace default. | Source will not emit events until channel exists. |
+
+\*: Required
+
+#### Status
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| channel | ObjectReference | The channel used to emit events. | |
+| Subscribable | Subscribable | Pointer to a channel which can be subscribed to in order to receive events from this source. | |
+| provisioned |[]ProvisionedObjectStatus| Creation status of each Channel and errors therein. | It is expected that a Source list all produced Channels. |
+
+##### Conditions
+
+ * **Ready.** True when the Source is provisioned and ready to emit events.
+ * **Provisioned.** True when the Source has been provisioned by a controller.
+
+#### Events
+
+ * Provisioned - describes each resource that is provisioned.
+
+### Life Cycle
+
+| Action | Reactions | Limitations |
+| --- | --- | --- |
+| Create | Provisioner controller watches for Sources and creates the backing resources depending on implementation. | |
+| Update | Provisioner controller synchronizes backing implementation on changes. | |
+| Delete | Provisioner controller will deprovision backing resources depending on implementation. | |
+
+---
+
+## kind: Channel
+
+### group: eventing.knative.dev/v1alpha1
+
+_A Channel logically receives events on its input domain and forwards them to
+its subscribers. Additional behavior may be introduced by using the
+Subscription's call parameter._
+
+### Object Schema
+
+#### Spec
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| provisioner*| ProvisionerReference | The name of the provisioner to create the resources that back the Channel. | Immutable. |
+| arguments | runtime.RawExtension (JSON object)| Arguments to be passed to the provisioner. | |
+| channelable | Channelable | Holds a list of downstream subscribers for the channel. | |
+| eventTypes |[]String| An array of event types that will be passed on the Channel. | Must be objects with kind:EventType. |
+
+\*: Required
+
+#### Metadata
+
+##### Owner References
+
+ * If the Pipeline controller created this Channel: Owned by the originating Pipeline.
+ * Owned by the Provisioner used to provision the Channel.
+
+#### Status
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| Sinkable | Sinkable | Address to the endpoint as top-level domain that will distribute traffic over the provided targets from inside the cluster. | |
```

Comment:
This is different than the [Sinkable](https://github.com/knative/eventing/blob/f7689d056b12298cce32eb949dd2b2623a0c6de9/docs/spec/interfaces.md#sinkable) definition, which says it should be status.targetable.domainInternal.